### ISSUE TITLE
LCFS - New Transfer page

### ISSUE THREAD

**Issue by justin-lepitzki** [bcgov/lcfs#212]
### Description:
Create the new "Transfer" button and corresponding landing page.

- **Button location:** On the Transactions page.
- **Button visibility:** Only visible to BCeID users with the "Transfers" role.
- **New transfers page contents:** Contains 'New Transfer' heading and some introductory text.

### Wireframe:
- New transfer button: https://preview.uxpin.com/59145260c7d8cc48969ac79401cc03d84843c3d9#/pages/164920204

- New Transfer page: https://preview.uxpin.com/59145260c7d8cc48969ac79401cc03d84843c3d9#/pages/164975114/simulate/no-panels

### Purpose and Benefit to User:
- This feature represents the first steps in initiating a new transfer, streamlining the process for users.

### Acceptance Criteria:
- Given I am a user with the "Transfers" role, when I visit the Transactions page, then I should see the "New Transfer" button.
- Given I am a user without the "Transfers" role, when I visit the Transactions page, then I should not see the "New Transfer" button.
- Given I am a user with the "Transfers" role, when I click on the "New Transfer" button, then I should be taken to the New Transfers page with the appropriate heading and text.

### Development Checklist:
- [ ] A. Implement the "New Transfer" button on the Transactions page.
- [ ] B. Ensure the button is only visible to BCeID users with the "Transfers" role.
- [ ] C. Develop the New Transfers landing page with the specified heading and text.

**Comment by justin-lepitzki** [comment]
@Kailee-Douglas and @airinggov can we update the wireframe to include a breadcrumb of page location. I believe the breadcrumb should read Home > Transactions > New Transfer

**Comment by justin-lepitzki** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Low-Carbon-Fuel-Standard-644c2db3be098a001977ad0b/issues/bcgov/lcfs/212?planning-poker) with Zenhub @AlexZorkin @hamed-valiollahi @jig-patel @kevin-hashimoto @prv-proton

**Comment by Kailee-Douglas** [comment]
@justin-lepitzki  Here are the updated mock-ups with the breadcrumbs. I referenced what you attached above, but they are now with the LCFS header.

https://preview.uxpin.com/59145260c7d8cc48969ac79401cc03d84843c3d9#/pages/164920204

https://preview.uxpin.com/59145260c7d8cc48969ac79401cc03d84843c3d9#/pages/164920192

**Comment by justin-lepitzki** [comment]
@Kailee-Douglas Looks good! A couple tweaks on this [wireframe]( https://preview.uxpin.com/59145260c7d8cc48969ac79401cc03d84843c3d9#/pages/164920204): The heading should be Transactions and the breadcrumb should be Home > Transactions

**Comment by Kailee-Douglas** [comment]
Of course, thanks @justin-lepitzki  It's been updated: https://preview.uxpin.com/59145260c7d8cc48969ac79401cc03d84843c3d9#/pages/164920204

**Comment by Grulin** [comment]
Requires BCeID user access to test.

**Comment by Grulin** [comment]
As a BCeID user with only read-only permissions, I experience the following:
1) The "New transaction" label should say "New Transfer" instead.
2) With read-only permissions, this "New transaction" button should not be visible to me. This button should only be visible via the Transactions tab to those BCeID users with transfer permissions. 
3) I get the following error when first landing on the Transactions page, but am still able to press "New transaction" button and navigate to that page. Please see the following screenshot:



![Image](https://github.com/bcgov/lcfs/assets/156380046/9b22a12a-3f34-49fc-98c8-2b7e87f1b8ba)

**Comment by hamed-valiollahi** [comment]
@Grulin, thanks for pointing out these issues.

1. I've updated the button's label. (The pull request is created)
2. I've looked into the code and noticed that the new transfer button shows up for users with 'Supplier' or 'Transfer' roles. Even if a user has other roles like 'Read Only', they'll still see the button if they have 'Supplier' or 'Transfer' roles.
3. Thanks for the screenshot. It looks like the server is giving a 403 (Forbidden) error, which means you don't have the required roles to view this table. From what I understand, we'll fix this issue later. We need to do a detailed check of roles.

**Comment by Grulin** [comment]
Sounds good. 'Supplier' shouldn't be a role, but it sounds like this is being dealt with in review elsewhere. The roles on the front end are only manage users, transfer, compliance reporting and signing authority. I don't know if supplier is still used on the developer side maybe?  Thank you for this, @hamed-valiollahi

**Comment by hamed-valiollahi** [comment]
@Grulin Sorry! you're right, Lindsy! As we discussed, below are the roles we have in the database:

The first two roles, GOVERNMENT and SUPPLIER, are special because GOVERNMENT role can be considered as IDIR user, and SUPPLIER role can be considered as BCeID user. These two are considered general roles.

```
* GOVERNMENT
* SUPPLIER
ADMINISTRATOR
ANALYST
COMPLIANCE_MANAGER
DIRECTOR
MANAGE_USERS
TRANSFER
COMPLIANCE_REPORTING
SIGNING_AUTHORITY
READ_ONLY
```

I've updated the code so that users with the TRANSFER role can see the 'New Transfer' button.
